term/fhir-term/src/main/java/com/ibm/fhir/term/util/CodeSystemSupport.java

```diff
@@ -232,6 +247,124 @@ public static CodeSystem getCodeSystem(java.lang.String url) {
         return FHIRRegistry.getInstance().getResource(url, CodeSystem.class);
     }
 
+    /**
+     * @param codeSystem
+     * @return
+     *    true if the passed codeSystem has a property-based concept hierarchy
+     */
+    public static boolean hasPropertyHierarchy(CodeSystem codeSystem) {
+        for (CodeSystem.Property p : codeSystem.getProperty()) {
+            Uri uri = p.getUri();
+            if (uri != null && PARENT_CHILD_PROPS.contains(uri.getValue())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Convert the passed codeSystem into a "simple" CodeSystem...one where all concept hierarchy is expressed
+     * through nested concepts rather than properties.
+     *
+     * @param codeSystem
+     * @return
+     *    the code system associated with the given input parameter, or null if no such code system exists
+     * @implSpec for CodeSystems with no parent or child properties, this will return the same CodeSystem that was passed
+     */
+    public static CodeSystem convertToSimpleCodeSystem(CodeSystem codeSystem) {
+        if (!hasPropertyHierarchy(codeSystem)) {
+            return codeSystem;
+        }
+//        for (Concept c : codeSystem.getConcept()) {
```

Comment:
remove dead code

Reply:
done.  initially I was thinking of throwing in cases where a CodeSystem has a mix of nested concepts and property-based hiearchy.  however, I ended up adding a nested concept to my test codesystems and it seemed to work ok (it just won't look for the properties on the nested concepts).

I've added the following implNote to explain this:
> In cases where the CodeSystem has both nested concepts AND property-based concepts, only the top-level concepts will be checked for properties. It is not recommended to mix hierarchy approaches like this.